### PR TITLE
Composing history and pipeline export corrections

### PR DESCRIPTION
**Pull request by nicl-nno**
The modifications of composing history and pipeline export for better compability with Fedot.Web

### CODE REVIEW

**Review by MAGLeb — CHANGES_REQUESTED** [review]

File: fedot/core/pipelines/template.py [hunk 1]
```diff
@@ -80,45 +82,56 @@ def _extract_pipeline_structure(self, node: Node, operation_id: int, visited_nod
             self.operation_templates.append(operation_template)
             self.total_pipeline_operations[operation_template.operation_type] += 1
 
-    def export_pipeline(self, path: str):
+    def export_pipeline(self, path: str = None, root_node: Node = None):
         """
         Save JSON to path and return this JSON like object.
         :param path: custom path to save
         :return: JSON like object
+        :root_node: root node of exported pipeline
```
[the remark this answers is not in the document]
Comment (nicl-nno) [reply fedot/core/pipelines/template.py]: [Добавил](https://github.com/nccr-itmo/FEDOT/pull/372/files#diff-f4b05f3f14ea72c656a09e4098af6b9461d58bd93aae50477030a08921d477fbR85-R91).

File: fedot/api/api_utils.py [hunk 2]
```diff
@@ -175,7 +180,18 @@ def compose_fedot_model(train_data: [InputData, MultiModalData],
                                validation_blocks=validation_blocks,
                                timeout=datetime.timedelta(minutes=timeout_for_composing))
 
-    optimizer_parameters = GPGraphOptimiserParameters(genetic_scheme_type=GeneticSchemeTypesEnum.parameter_free)
+    genetic_scheme_type = GeneticSchemeTypesEnum.parameter_free
+
+    if 'steady_state' in 'preset':
```
[the remark this answers is not in the document]
Comment (nicl-nno) [reply fedot/api/api_utils.py]: [Изменил](https://github.com/nccr-itmo/FEDOT/pull/372/files#diff-0975751fd631b03bfdbe49d411b336f269e14dec2d1a4a652eaf42698070ae0eR184-R185).

File: fedot/core/pipelines/template.py [hunk 3]
```diff
@@ -80,45 +82,56 @@ def _extract_pipeline_structure(self, node: Node, operation_id: int, visited_nod
...
 
         if not os.path.exists(absolute_path):
             os.makedirs(absolute_path)
 
-        pipeline_template_dict = self.convert_to_dict()
-        json_data = json.dumps(pipeline_template_dict, indent=4)
-
         with open(os.path.join(absolute_path, f'{self.unique_pipeline_id}.json'), 'w', encoding='utf-8') as f:
-            f.write(json_data)
+            f.write(json.dumps(json.loads(json_data), indent=4))
```
[the remark this answers is not in the document]
Comment (nicl-nno) [reply fedot/core/pipelines/template.py]: Переделал [так](https://github.com/nccr-itmo/FEDOT/pull/372/files#diff-f4b05f3f14ea72c656a09e4098af6b9461d58bd93aae50477030a08921d477fbR105-R108)

File: fedot/core/optimisers/gp_comp/gp_optimiser.py [hunk 4]
```diff
@@ -28,6 +28,15 @@
 MIN_POPULATION_SIZE_WITH_ELITISM = 2
 
 
+class SimpleArchive(list):
```
[the remark this answers is not in the document]
Comment (nicl-nno) [reply fedot/core/optimisers/gp_comp/gp_optimiser.py]: Перенес в archive.py

File: fedot/core/pipelines/template.py [hunk 5]
```diff
@@ -139,15 +152,22 @@ def _prepare_paths(self, path: str):
...
+        if type(source) is str:
+            path = source
+            self._check_path_correct(path)
 
-        with open(path) as json_file:
-            json_object_pipeline = json.load(json_file)
-            self.log.message(f"The pipeline was imported from the path: {path}.")
+            with open(path) as json_file:
+                json_object_pipeline = json.load(json_file)
+                self.log.message(f'The chain was imported from the path: {path}.')
```
[the remark this answers is not in the document]
Comment (nicl-nno) [reply fedot/core/pipelines/template.py]: Поменял.

**Review by MAGLeb — APPROVED** [review]

**Pull request merged**